client/lobby/SelectionTab.cpp

```diff
@@ -868,7 +927,28 @@ void SelectionTab::parseSaves(const std::unordered_set<ResourcePath> & files)
 		}
 		catch(const std::exception & e)
 		{
-			logGlobal->error("Error: Failed to process %s: %s", file.getName(), e.what());
+			// asking for deletion of unsupported saves
+			if(CSH->isHost())
+			{
+				if(deleteUnsupported == DELETE)
+					doDeleteUnsupported(file.getName());
+				else if(deleteUnsupported == NONE)
+				{
+					CInfoWindow::showYesNoDialog(CGI->generaltexth->translate("vcmi.lobby.deleteUnsupportedSave"), std::vector<std::shared_ptr<CComponent>>(), [&deleteUnsupported, doDeleteUnsupported, file](){
+						doDeleteUnsupported(file.getName());
+						deleteUnsupported = DELETE;
+					}, [&deleteUnsupported](){ deleteUnsupported = REMAIN; });
+
+					while(deleteUnsupported == NONE)
+					{
+						auto unlockInterface = vstd::makeUnlockGuard(GH.interfaceMutex);
+						boost::this_thread::sleep_for(boost::chrono::milliseconds(5));
+					}
```

Comment:
Please avoid such busy-waits. Better to generate list of saves & process it later